vectorsearch/resource_vector_search_endpoint.go

```diff
@@ -47,6 +48,13 @@ func ResourceVectorSearchEndpoint() common.Resource {
 			}
 			endpoint, err := wait.GetWithTimeout(d.Timeout(schema.TimeoutCreate))
 			if err != nil {
+				ctx2, cancel := context.WithDeadline(context.Background(), time.Now().Add(10*time.Second))
```

Comment:
instead of ctx2, we could probably use another variable, maybe: `ctxWithDeadline`